layers.go

```diff
@@ -1988,10 +2037,20 @@ func (r *layerStore) deleteInDigestMap(id string) {
 }
 
 // Requires startWriting.
-func (r *layerStore) Delete(id string) error {
+// This is soft-deprecated and should not have any new callers; use deferredDelete instead.
+func (r *layerStore) deleteWhileHoldingLock(id string) error {
+	cleanupFunctions, deferErr := r.deferredDelete(id)
+	cleanupErr := tempdir.CleanupTemporaryDirectories(cleanupFunctions...)
+	return errors.Join(deferErr, cleanupErr)
+}
+
+// Requires startWriting.
+// Caller MUST run all returned cleanup functions after this. Ideally outside of
```

Comment:
> // Caller MUST run all returned cleanup functions after this.

Reading how we nave `cleanFunctions` set, and then we can fail in `saveFor`, I think this should explicitly document “EVEN IF the function returns an error”. Some of the callers already do this, I didn’t check whether all do.